src/base/core/int.c

```diff
@@ -1465,6 +1467,16 @@ static int msdos(void)
 		    redir_state++;
 	    }
 
+            /*
+             * Let's set up the hma start address for dosdebug use. There's no
+             * specific reason it should be here other than it needs to be done
+             * sometime after DOS has started.
```

Comment:
`dos_post_boot()` please.
Also if init wasn't called (yet), dosdebug
should try from HMA start.

Reply:
I moved the startup logic to `dos_post_boot()`. Are you sure about dosdebug trying from HMA start if we haven't done the init yet, as we don't know actually know the start address (it varies with each DOS) and you wanted to avoid false positives with scanning from ffff:0000?

Reply:
What's the current behavior?
Not parse HMA at all?
That's fine.

Reply:
Yes if the hma_start variable is NULL, then hma is not parsed. Somewhere I need to initialise it to zero though, and I suppose set it to zero on reset?

Reply:
Yes, please init your stuff in
dos_post_boot_reset().